client/components/checklist/index.js

```diff
@@ -16,37 +16,53 @@ import TaskPlaceholder from 'components/checklist/task-placeholder';
 export default class Checklist extends PureComponent {
 	static propTypes = {
 		isPlaceholder: PropTypes.bool,
+		updateCompletion: PropTypes.func,
 	};
 
-	state = { hideCompleted: false };
-
-	toggleCompleted = () =>
-		this.setState( ( { hideCompleted } ) => ( { hideCompleted: ! hideCompleted } ) );
+	componentDidMount() {
+		this.notifyCompletion();
+	}
 
-	renderPlaceholder() {
-		return (
-			<div className={ classNames( 'checklist', 'is-expanded', 'is-placeholder' ) }>
-				<ChecklistHeader total={ 0 } completed={ 0 } />
-				<div className="checklist__tasks">
-					{ times( Children.count( this.props.children ), index => (
-						<TaskPlaceholder key={ index } />
-					) ) }
-				</div>
-			</div>
-		);
+	componentDidUpdate() {
+		this.notifyCompletion();
 	}
 
-	render() {
-		if ( this.props.isPlaceholder ) {
-			return this.renderPlaceholder();
+	notifyCompletion() {
+		if ( 'function' === typeof this.props.updateCompletion ) {
```

Comment:
Could provide a `noop` default prop instead. Might obfuscate things a bit tho.

(I guess I just don't like the check here too much, but total nitpick territory.)